tests/AbstractLoopTest.php

```diff
@@ -16,11 +16,39 @@ public function setUp()
 
     abstract public function createLoop();
 
+    /**
+     * @deprecated Use createSocketPair() instead.
+     */
     public function createStream()
     {
         return fopen('php://temp', 'r+');
     }
+    
+    public function isUnixSocketSupported()
+    {
+        return in_array('unix', stream_get_transports(), true);
+    }
 
+    public function createSocketPair()
+    {
+        if ($this->isUnixSocketSupported()) {
+            $sockets = stream_socket_pair(STREAM_PF_UNIX, STREAM_SOCK_STREAM, STREAM_IPPROTO_IP);
+        } else {
+            $sockets = stream_socket_pair(STREAM_PF_INET, STREAM_SOCK_STREAM, STREAM_IPPROTO_IP);
```

Comment:
I could turn this into one branch like this:
```php
$domain = $this->isUnixSocketSupported() ? STREAM_PF_UNIX : STREAM_PF_INET;
$sockets = stream_socket_pair($domain, STREAM_SOCK_STREAM, STREAM_IPPROTO_IP);
```
This ensures that it is allways the same function call.